### PR TITLE
test: add az fileshare storage integration tests with real az resources

### LINKED ISSUE #179: FileShare folder creation: HTTP 409 error when folder already existing
**Describe the bug**
When using Arcus.Scripting.Storage.FileShare with command Create-AzFileShareStorageFolder creation of the folder is successful. When I retrigger my pipeline (and the folder already exists) I get following error:
image

**To Reproduce**
-Execute an Azure Powershell step more than once in a release pipeline in DevOps using the Create-AzFileShareStorageFolder. When the folder already exists this error is thrown.

**Expected behavior**
I expected the command to check for the existance of the folder and if present: do nothing. If the folder is not present: create it.

**Additional context**
Azure Powershell version 5.185.0
Arcus version is latest I think. Imported using Install-Module -Name Arcus.Scripting.Storage.FileShare

### PR DESCRIPTION
**Pull request by stijnmoreels**
Add new integration tests for the Azure FileShare storage PS module.

Relates to #174 
Closes #179

### CODE REVIEW

File: src/Arcus.Scripting.Storage.FileShare/Scripts/Copy-AzFileShareStorageFiles.ps1 [hunk 1]
```diff
@@ -7,57 +7,39 @@ param(
...
-            {
-                ## Upload the file  
+$storageAccount = Get-AzStorageAccount -ResourceGroupName $ResourceGroupName -Name $StorageAccountName
+$context = $storageAccount.Context
+
+Write-Verbose "Upload files to Azure FileShare storage '$FileShareName'..."
+
+if (VerifyAzureFileShareExists) {
+    foreach ($file in Get-ChildItem ("$SourceFolderPath") -File) {
+        try {
```
Comment (stijnmoreels) [inline src/Arcus.Scripting.Storage.FileShare/Scripts/Copy-AzFileShareStorageFiles.ps1:32]: Moved the try/catch to the loop itself as it doesn't seem logical to stop the next file from uploading when the current file fails.

File: src/Arcus.Scripting.Storage.FileShare/Scripts/Create-AzFileShareStorageFolder.ps1 [hunk 2]
```diff
@@ -5,31 +5,16 @@ param(
...
-    Write-Error "Failed to create the directory '$FolderName' in file-share '$FileShareName'. Reason: $ErrorMessage"
-    return $null
-}
+if ($FolderName -in $fileShareFolders.Name) {
+    Write-Host "Azure FileShare storage folder '$FolderName' already exists, skipping"
+} else {
+    Write-Verbose "Creating Azure FileShare storage folder '$FolderName' in file share '$FileShareName'.."
+    New-AzStorageDirectory -Context $storageAccount.Context -ShareName $FileShareName -Path $FolderName
+    Write-Host "Created Azure FileShare storage folder '$FolderName' in file share '$FileShareName'"
+}
```
Comment (stijnmoreels) [inline src/Arcus.Scripting.Storage.FileShare/Scripts/Create-AzFileShareStorageFolder.ps1:20]: Removed the try/catch-match block to determine whether or not the folder already exists and instead getting the list of existing directories and find one that matches the provided name.

File: src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.Storage.FileShare.tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+                    -ResourceGroupName $config.Arcus.ResourceGroupName `
+                    -StorageAccountName $config.Arcus.Storage.StorageAccount.Name `
+                    -FileShareName $nonExistingFileShareName `
+                    -SourceFolderPath "$PSScriptRoot\Blobs" `
+                    -DestinationFolderName $folderName } |
+                    Should -Throw
+            }
+        }
+        AfterEach {
+            Remove-AzStorageShare -Name $fileShareName -Context $storageAccount.Context -IncludeAllSnapshot -Force
```
Comment (stijnmoreels) [inline src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.Storage.FileShare.tests.ps1:95]: Removing the entire file share at each test makes sure that we don't have to remove every single file/directory ourselves.

File: src/Arcus.Scripting.Storage.FileShare/Scripts/Create-AzFileShareStorageFolder.ps1 (hunk 2 above)
Comment (mbraekman) [reply src/Arcus.Scripting.Storage.FileShare/Scripts/Create-AzFileShareStorageFolder.ps1:20]: Nice idea!

File: src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.Storage.FileShare.tests.ps1 (hunk 3 above)
Comment (mbraekman) [reply src/Arcus.Scripting.Tests.Integration/Arcus.Scripting.Storage.FileShare.tests.ps1:95]: Is the better/quicker option indeed. 
Didn't think of that :)

**Review by mbraekman — APPROVED** [review]

**Pull request merged**